packages/nuxt/src/core/nitro.ts

```diff
@@ -378,7 +378,7 @@ export async function initNitro (nuxt: Nuxt & { _nitro?: Nitro }) {
 
   // Trigger Nitro reload when SPA loading template changes
   nuxt.hook('builder:watch', async (_event, path) => {
-    if (normalize(path) === spaLoadingTemplatePath(nuxt)) {
+    if (normalize(path) === await spaLoadingTemplatePath(nuxt)) {
```

Comment:
Maybe move this outside of the hook so it doesn't rerun every time a file is modified?